Misc/NEWS.d/next/Core and Builtins/2023-07-27-11-47-29.gh-issue-104432.oGHF-z.rst

```diff
@@ -0,0 +1,4 @@
+Fix potential unaligned memory access on C APIs involving returned sequences
+of `char *` pointers within the :mod:`grp` and :mod:`socket` modules. These
+were revealed using a ``-fsaniziter=alignment`` build on ARM macOS. Patch by
```

Comment:
To clarify, I observed this on macOS for Intel. I have not confirmed if this also occurs on macOS for ARM, but I would not be surprised if it still does.

Reply:
Oops I shouldn't have bothered mentioning the architecture. no big deal though (feel free to make an edit PR if it bothers you).

I chose the word "revealed" in the wording as a way to indicate _how_ it was found which is more interesting.  I suspect the alignment sanitizer may flag this regardless of compilation target architecture but I didn't try to reproduce it myself. :)

I expect the fixup applies all over the place. Regardless of native transparent hardware support (x86) or triggering a trap to the OS to fix the unaligned access up (Alpha & a pile of very RISC thing) they are usually less efficient - so directly using code that detects it and does it right rather than triggering an even slower path is good.